src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs

```diff
@@ -22,16 +23,36 @@ public static class HalLearnersCatalog
         /// <param name="labelColumn">The labelColumn column.</param>
         /// <param name="featureColumn">The features column.</param>
         /// <param name="weights">The weights column.</param>
-        /// <param name="advancedSettings">Algorithm advanced settings.</param>
         public static OlsLinearRegressionTrainer OrdinaryLeastSquares(this RegressionContext.RegressionTrainers ctx,
             string labelColumn = DefaultColumnNames.Label,
             string featureColumn = DefaultColumnNames.Features,
-            string weights = null,
-            Action<OlsLinearRegressionTrainer.Arguments> advancedSettings = null)
+            string weights = null)
+        {
+            Contracts.CheckValue(ctx, nameof(ctx));
+            var env = CatalogUtils.GetEnvironment(ctx);
+            var options = new OlsLinearRegressionTrainer.Options
+            {
+                LabelColumn = labelColumn,
+                FeatureColumn = featureColumn,
+                WeightColumn = weights != null ? Optional<string>.Explicit(weights) : Optional<string>.Implicit(DefaultColumnNames.Weight)
+            };
+
+            return new OlsLinearRegressionTrainer(env, options);
+        }
+
+        /// <summary>
+        /// Predict a target using a linear regression model trained with the <see cref="OlsLinearRegressionTrainer"/>.
```

Comment:
>Predict [](start = 12, length = 7)

It doesn't look like this function accepts a feature vector and then produce a float.

Reply:
it can accept a feature vector in the options. 

---
In reply to: [248869365](https://github.com/dotnet/machinelearning/pull/2170#discussion_r248869365) [](ancestors = 248869365)